#project hardware/qcom-caf/msm8952/display/

diff --git a/libgralloc/alloc_controller.cpp b/libgralloc/alloc_controller.cpp
index 082558c..eb214fb 100755
--- a/libgralloc/alloc_controller.cpp
+++ b/libgralloc/alloc_controller.cpp
@@ -448,8 +448,10 @@ int IonController::allocate(alloc_data& data, int usage)
         ionHeapId |= ION_HEAP(ION_SYSTEM_HEAP_ID);
     }
 
+/* comment out ionHeapId = 0x100000 - ION_IOC_ALLOC failed with error - No such device
     if(usage & GRALLOC_USAGE_PRIVATE_CAMERA_HEAP)
         ionHeapId |= ION_HEAP(ION_CAMERA_HEAP_ID);
+*/
 
     if(usage & GRALLOC_USAGE_PRIVATE_ADSP_HEAP)
         ionHeapId |= ION_HEAP(ION_ADSP_HEAP_ID);
diff --git a/libhwcomposer/hwc.cpp b/libhwcomposer/hwc.cpp
index f47fe50..91cbce0 100644
--- a/libhwcomposer/hwc.cpp
+++ b/libhwcomposer/hwc.cpp
@@ -1023,7 +1023,7 @@ void hwc_dump(struct hwc_composer_device_1* dev, char *buff, int buff_len)
     dumpsys_log(aBuf, ovDump);
     ovDump[0] = '\0';
     dumpsys_log(aBuf, "Copybit::isAbcInUse=%d\n\n",isAbcInUse(ctx) ? 1 : 0);
-    strlcpy(buff, aBuf.string(), buff_len);
+    strlcpy(buff, aBuf.c_str(), buff_len);
 }
 
 int hwc_getActiveConfig(struct hwc_composer_device_1* dev, int disp)
diff --git a/libhwcomposer/hwc_dump_layers.cpp b/libhwcomposer/hwc_dump_layers.cpp
index 92e8caa..c3358dd 100644
--- a/libhwcomposer/hwc_dump_layers.cpp
+++ b/libhwcomposer/hwc_dump_layers.cpp
@@ -211,7 +211,7 @@ void HwcDebug::logHwcProps(uint32_t listFlags)
                 "[???]": "");
     }
     ALOGI("Display[%s] Layer[*] %s-HwcModuleCompType, %d-layer MdpComp %s",
-         mDisplayName, hwcModuleCompTypeLog.string(), sMdpCompMaxLayers,
+         mDisplayName, hwcModuleCompTypeLog.c_str(), sMdpCompMaxLayers,
         (listFlags & HWC_GEOMETRY_CHANGED)? "[HwcList Geometry Changed]": "");
 }
 
@@ -252,7 +252,7 @@ void HwcDebug::logLayer(size_t layerIndex, hwc_layer_1_t hwLayers[])
         sourceCrop.right, sourceCrop.bottom,
         displayFrame.left, displayFrame.top,
         displayFrame.right, displayFrame.bottom,
-        hwcVisRegsScrLog.string());
+        hwcVisRegsScrLog.c_str());
     // Log Line 2
     ALOGI("Display[%s] Layer[%zu] LayerCompType = %s, Format = %s, "
         "Orientation = %s, Flags = %s%s%s, Hints = %s%s%s, "
diff --git a/libhwcomposer/hwc_mdpcomp.cpp b/libhwcomposer/hwc_mdpcomp.cpp
index 020faf6..a3b0a42 100644
--- a/libhwcomposer/hwc_mdpcomp.cpp
+++ b/libhwcomposer/hwc_mdpcomp.cpp
@@ -2381,7 +2381,7 @@ int MDPComp::prepare(hwc_context_t *ctx, hwc_display_contents_1_t* list) {
                 (list->flags & HWC_GEOMETRY_CHANGED));
         android::String8 sDump("");
         dump(sDump, ctx);
-        ALOGD("%s",sDump.string());
+        ALOGD("%s",sDump.c_str());
     }
 
 #ifdef DYNAMIC_FPS
