diff --git a/core/version_defaults.mk b/core/version_defaults.mk
index a25aea3d1e..4377972309 100644
--- a/core/version_defaults.mk
+++ b/core/version_defaults.mk
@@ -240,7 +240,7 @@ ifndef PLATFORM_SECURITY_PATCH
     #  It must be of the form "YYYY-MM-DD" on production devices.
     #  It must match one of the Android Security Patch Level strings of the Public Security Bulletins.
     #  If there is no $PLATFORM_SECURITY_PATCH set, keep it empty.
-      PLATFORM_SECURITY_PATCH := 2024-02-05
+      PLATFORM_SECURITY_PATCH := 2025-03-05
 endif
 .KATI_READONLY := PLATFORM_SECURITY_PATCH
 
